src/diagrams/sequence/deal-flow.mmd

```diff
@@ -0,0 +1,35 @@
+sequenceDiagram
+participant StorageClient
+participant StorageProvider
+participant StorageMiningSubsystem
+participant StorageMarketActor
+StorageClient -->> StorageMarketActor: AddBalance()
+StorageProvider -->> StorageMarketActor: AddBalance()
+StorageClient -->+ StorageProvider: OnNewStorageDealProposal(proposal, payloadCID)
+StorageProvider -->- StorageClient: NotifyOfStorageDealProposalStatus(ProposalRejected)
+StorageClient -->+ StorageProvider: OnNewStorageDealProposal(proposal, payloadCID)
+StorageProvider -->- StorageClient: NotifyOfStorageDealProposalStatus(ProposalAccepted)
+StorageProvider -->+ StorageClient: graphsync.Pull(payloadCID)
+StorageClient -->- StorageProvider: graphsync.Push(payload IPLD)
+StorageProvider -->+ StorageProvider: OnReceivingPayload(payloadCID)
+StorageProvider -->> StorageProvider: verifyStorageDealProposal(proposal)
+StorageProvider -->> StorageProvider: verifyPieceCID(proposal, payloadCID)
+StorageProvider -->> StorageProvider: signStorageDealProposal(proposal)
+StorageProvider -->- StorageClient: NotifyOfStorageDealProposalStatus(ProposalSigned)
+StorageProvider -->> StorageProvider: publishStorageDealMessage(message)
+StorageProvider -->+ StorageMarketActor: PublishStorageDeal(message)
+StorageMarketActor -->> StorageMarketActor: generateStorageDealID()
+StorageMarketActor -->> StorageMarketActor: lockFund()
+StorageMarketActor -->- StorageProvider: dealID
+StorageProvider -->> StorageClient: NotifyOfStorageDealStatus(DealPublished)
+StorageProvider -->> StorageMiningSubsystem: HandleStorageDeal(deal)
+StorageMiningSubsystem -->> StorageMinerActor: CommitSector()
+StorageProvider -->> StorageClient: NotifyOfStorageDealStatus(DealCommitted)
+CronActor -->> StorageMinerActor: NotifyOfPoStChallenge()
+StorageMiningSubsystem -->> StorageMinerActor: SubmitPoSt()
+StorageMinerActor -->> StorageMinerActor: activateSector
```

Comment:
This feels like it should be a call on the Market actor (called from miner.CommitSector.

For reference this is what lotus currently does:
https://github.com/filecoin-project/lotus/blob/devnet/6/chain/actors/actor_miner.go#L298

https://github.com/filecoin-project/lotus/blob/devnet/6/chain/actors/actor_storagemarket.go#L427

Reply:
Ah, this is something separate, this part refers to a sector being counted towards power after the first successful SubmitPoSt and hence Active in SectorState.